apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.cs

```diff
@@ -62,6 +62,15 @@ internal SubscriberClientImpl(SubscriptionName subscriptionName, IEnumerable<Sub
         _modifyDeadlineSeconds = (int)((settings.AckDeadline ?? DefaultAckDeadline).TotalSeconds);
         var autoExtendInterval = TimeSpan.FromSeconds(_modifyDeadlineSeconds) - (settings.AckExtensionWindow ?? DefaultAckExtensionWindow);
         var autoExtendIntervalForExactlyOnceDelivery = settings.AckExtensionWindow ?? MinimumAckExtensionWindowForExactlyOnceDelivery;
+        // The value of 15 seconds ensures that we get ample time before the Ack deadline expires and that we don't send too frequent lease extensions.
+        var preferredExtendInterval = TimeSpan.FromSeconds(_modifyDeadlineSeconds - 15);
```

Comment:
And now
preferredExtendInterval => is a delay before we can attempt to extend the lease.